.github/workflows/KubernetesExternalClients.yaml

```diff
@@ -43,43 +43,67 @@ jobs:
-        export_default_credentials: true 
+        credentials_json: '${{ secrets.GKE_SA_KEY }}'
+
+    - name: 'Set up Cloud SDK'
+      uses: 'google-github-actions/setup-gcloud@v1'
 
     - name: Install Kubectl
       run: |-
         gcloud components install kubectl
+
+    - name: Install Helm
+      uses: azure/setup-helm@v3
+      id: install
+      
     - name: Create GKE cluster
       run: |-
         repoName='kubernetes-clients-compatibility'
         clusterName="$repoName-$GITHUB_RUN_NUMBER"
         echo "clusterName=$clusterName" >> $GITHUB_ENV
-        gcloud container clusters create "$clusterName" --zone="$GKE_ZONE" --project="$PROJECT_ID" --machine-type=n1-standard-2 --num-nodes=2 --no-enable-ip-alias
+        gcloud container clusters create "$clusterName" \
+        --zone="$GKE_ZONE" \
+        --project=${{ secrets.PROJECT_ID }} \
+        --machine-type=n1-standard-2 \
+        --num-nodes=2 \
+        --network=${{env.GCP_NETWORK}} 
         sleep 30
+
     - name: Deploy Hazelcast cluster
       run: |-
-        kubectl apply -f https://raw.githubusercontent.com/hazelcast/hazelcast-kubernetes/master/rbac.yaml
-        kubectl create service loadbalancer hz-hazelcast-0 --tcp=5701
-        kubectl run hz-hazelcast-0 --image=hazelcast/hazelcast:$HAZELCAST_VERSION --port=5701 -l "app=hz-hazelcast-0,role=hazelcast"
-        kubectl create service loadbalancer hz-hazelcast-1 --tcp=5701
-        kubectl run hz-hazelcast-1 --image=hazelcast/hazelcast:$HAZELCAST_VERSION --port=5701 -l "app=hz-hazelcast-1,role=hazelcast"
-        kubectl create service loadbalancer hz-hazelcast-2 --tcp=5701
-        kubectl run hz-hazelcast-2 --image=hazelcast/hazelcast:$HAZELCAST_VERSION --port=5701 -l "app=hz-hazelcast-2,role=hazelcast"
-        kubectl create service loadbalancer hz-hazelcast --tcp=5701 -o yaml --dry-run=client | kubectl set selector --local -f - "role=hazelcast" -o yaml | kubectl create -f -
-    - name: Wait for deployment to finish
-      run: |-
-        kubectl wait --for=condition=ready pod/hz-hazelcast-0 --timeout=180s
-        kubectl wait --for=condition=ready pod/hz-hazelcast-1 --timeout=180s
-        kubectl wait --for=condition=ready pod/hz-hazelcast-2 --timeout=180s
+        helm repo add hazelcast https://hazelcast-charts.s3.amazonaws.com/
+        helm repo update
+        helm install operator hazelcast/hazelcast-platform-operator --set installCRDs=true       
+        cat <<EOF | kubectl apply -f -
+        apiVersion: hazelcast.com/v1alpha1
+        kind: Hazelcast
+        metadata:
+          name: hz-hazelcast
+        spec:
+          clusterSize: 3
+          exposeExternally:
+            type: Smart
+            discoveryServiceType: LoadBalancer
+            memberAccess: LoadBalancer
+        EOF
+
+    - name: Wait for deployment to be running
+      run: |-                    
+        while [ "$(kubectl get hazelcast -o jsonpath='{.items[*].status.phase}')" != "Running" ]; do
```

Comment:
You make 2 `kubectl get` calls one here and one when you print it and they may not be the same if you are lucky. Hence, I suggest you make one call and print that only, You may use `do while` loop instead.

Reply:
What will happened if they are not same? It just prints status. If it's ready in first round (which is very very unlikely), loop will be passed. If while condition is met but status is changed while printing, it doesn't contradicts with the context. It will print the status.